Runtime image set as environment variable

Artyop opened the pull request:
This PR sets the runtime-image as environment variable, and fixes the runtime image update script 
Please check commit description for more details

What's needed : build the new runtime image and push it ahead of merging this PR, because the runtime image update script is part of the runtime directory, whose checksum is used to generate the tag.

```release-note
Runtime image set as environment variable
```

Artyop committed:
refac: extract runtime image to an env var

This commit extract the runtime image referenced in workflow build-image-ci to a environment variable referenced in the action setting up environment variables.
The variable will be able to be override without updating the dockerfiles nor the worfklow build-image-ci.

Signed-off-by: Antony Reynaud <antony.reynaud@isovalent.com>

Artyop committed:
refac: update runtime image

This commit refac the sed commands :
- for Dockerfile, it will not check what is prefixed and will update the image based on its name, so if any changes are name for any reason on prefixed image name on Dockerfiles this won't disrupt the image tag/digest update.
    example :
        if CILIUM_RUNTIME_IMAGE=quay.io/cilium/cilium-runtime:tag@digest
        is changed for RUNTIME_IMAGE=quay.io/cilium/cilium-runtime:tag@digest (or any other name)
        this won't affect the update process
- for workflows, since the runtime image has been moved to an action, we now look on the .github/actions directory. Also added a double quote character to be sure no templating issue will be encountered when updating the set-env action.

It also changes the way the image is made. Before that the image was a constant set in the script, now it will be extracted from the image_full variable passed as parameter of the script.
This will allow us to be able to switch repository without having to update the update cilium runtime image script.

Signed-off-by: Antony Reynaud <antony.reynaud@isovalent.com>

aanm commented on .github/workflows/build-images-ci.yaml:
```diff
@@ -237,6 +238,7 @@ jobs:
           target: release
           build-args: |
+            BASE_IMAGE=${{ env.CILIUM_RUNTIME_IMAGE }}
             MODIFIERS="NOSTRIP=1"
```
CILIUM_RUNTIME_IMAGE=${{ env.CILIUM_RUNTIME_IMAGE }}

Artyop committed:
refac: lint update runtime image

The script update-cilium-runtime-image.sh from scripts directory is used to lint cilium runtime image in CI.
This commit removes the image name from constant variable in update-cilium-runtime-image.sh script and allows us to pass a different image in parameter,
keeping quay.io/cilium/cilium-runtime as default parameter in the makefile, if nothing is provided to the script/target.
In the lint-images-base CI workflow it then passes CILIUM-RUNTIME-IMAGE environment variable as parameter.
This will allow us to change the repository of the runtime image without having to change the whole linting process.

Signed-off-by: Antony Reynaud <antony.reynaud@isovalent.com>

aanm committed:
images: update cilium-{runtime,builder}

Signed-off-by: André Martins <andre@cilium.io>
